core/tests/protractor/embedding.js

```diff
@@ -170,24 +248,23 @@ describe('Embedding', function() {
       browser.switchTo().defaultContent();
     };
 
-    users.createUser('embedder2@example.com', 'Embedder2');
+    users.createModerator('embedder2@example.com', 'Embedder2');
     users.login('embedder2@example.com', true);
-    adminPage.reloadExploration('protractor_test_1.yaml');
 
     // Change language to Thai, which is not a supported site language.
-    general.openEditor('12');
+    general.openEditor(explorationId);
     editor.setLanguage('ภาษาไทย');
     editor.saveChanges('Changing the language to a not supported one.');
     // We expect the default language, English
     checkPlaceholder('Type a number');
 
     // Change language to Spanish, which is a supported site language.
-    general.openEditor('12');
+    general.openEditor(explorationId);
     editor.setLanguage('español');
     editor.saveChanges('Changing the language to a supported one.');
     checkPlaceholder('Ingresa un número');
 
     users.logout();
-    general.checkForConsoleErrors([]);
+    general.checkForConsoleErrors(['.*idToBeReplaced.*']);
```

Comment:
The added element in the array should not be needed. It may cause other legitimate errors to not be displayed. If you only show stuff when the exploration ID is specified, will that solve the issue?

Reply:
hiding the element by setting display to 'none' doesn't help, as it still gives a console error.
and as 'idToBeReplaced' is invalid exp id so it will give the error one time.
so can you please suggest how should i go about this. 

I tried adding [these](https://github.com/GanitGenius/oppia/blob/2f0d4df9e5e2aa9561f17c0967aed6d4f9c6288d/assets/scripts/embedding_tests_dev_0.0.1.min.html#L9) scripts dynamically but it uses window.onload(), so can we change that to load it anytime that script is added??

Reply:
Ah, I see the issue. In Angular-land we'd just use ng-if, but no such luck here.

I think we can leave the console error check exception here, but let's use the exact text of the error rather than a general regex, and add a comment above the line explaining why it is necessary so that devs know that this is a special case.

Reply:
done!!